hyperspace/solana/src/client.rs

```diff
@@ -1103,6 +1109,72 @@ pub async fn get_accounts(
 	}
 }
 
+/// Parses additional accounts from the memo of a transaction directed at the bridge escrow.
+///
+/// Memo is comma separated list of the form
+/// `N,account-0,account-1,...,account-N-1,intent-id,embedded-memo`.  Embedded
+/// memo can contain commas.
+/// Returns a list of `account-i` for i=0..N or `None`
+/// if the memo does not conform to this format.  Note that no validation on
+/// accounts is performed.
+fn parse_intent_memo_accounts(memo: &str) -> Option<Vec<AccountMeta>> {
+	let (count, mut rest) = memo.split_once(',')?;
+	let count = count.parse::<usize>().ok()?;
+	let mut strings = rest.splitn(count + 2, ',').collect::<Vec<_>>();
+	// only accept a memo where all the components are presented
+	if strings.len() < count + 2 {
+		return None;
+	}
+	// take first `count` entries
+	let accounts = strings
+		.drain(..count)
+		.map(Pubkey::from_str)
+		.map(Result::ok)
+		.collect::<Option<Vec<Pubkey>>>()?;
+	let account_metas = accounts
+		.into_iter()
+		.map(|pubkey| AccountMeta { pubkey, is_signer: false, is_writable: true })
+		.collect::<Vec<AccountMeta>>();
+	Some(account_metas)
```

Comment:
The intermediary vector can be avoided quite easily.  Also, in this
instance you don’t want to use Vec::drain because that needs to resize
the source vector.

```suggestion
	strings
		.iter()
		.take(count)
		.map(|key| Pubkey::from_str(key).ok().map(|pubkey| AccountMeta {
			pubkey,
			is_signer: false,
			is_writable: true,
		})
		.collect()
```

(The `strings` vector could also be avoided but that’s a bit more code so probably not worth in the relayer).

Reply:
Fixed